tests/inverted-index-test.js

```diff
@@ -1,56 +1,262 @@
+    expect(invertedIndexObject.searchIndex(invertedIndexObject.index, 'book2.json', 'Set')).toEqual({ set: [0, 1] });
+  });
+
+  it('should ensure searchIndex goes through all indexed files if a fileName is not provided', () => {
+    const result = { 'book2.json': { from: [1], understand: [0, 1] }, 'book1.json': { from: [-1], understand: [-1] } };
+    expect(invertedIndexObject.searchIndex(invertedIndexObject.index, ['understand', 'from'])).toEqual(result);
+  });
+
+  it('should ensure searchIndex handles varied number of search terms', () => {
+    expect(invertedIndexObject.searchIndex(invertedIndexObject.index, 'book2.json', 'to', 'is', ['into', 'problem'])).toEqual({ to: [0, 1], is: [1], into: [0], problem: [0, 1] });
+  });
+
+  it('should ensure index returns correct result when searched', () => {
+    expect(invertedIndexObject.searchIndex(invertedIndexObject.index, 'book1.json', 'catwalk')).toEqual({ catwalk: [-1] });
+  });
+
+  it('should ensure a valid index is passed in', () => {
+    expect(invertedIndexObject.searchIndex('My index', 'book1.json', 'catwalk')).toEqual({ error: 'A valid index has not been created. Kindly create index before searching' });
+  });
+
+  it('should ensure searchIndex handles an array of search terms', () => {
+    expect(invertedIndexObject.searchIndex(invertedIndexObject.index, 'book1.json', ['accept', 'the'])).toEqual({ accept: [1], the: [0, 1] });
+  });
+
+  it('should return proper error message if index has not been created for the provided filename', () => {
+    expect(invertedIndexObject.searchIndex(invertedIndexObject.index, 'book4.json', 'third')).toEqual({ error: 'Index has not been created for the specified file' });
+  });
+});
+
+describe('Validity of JSON array', () => {
+  it('should return true for a valid JSON array', () => {
+    expect(invertedIndexObject.isValidJSON(invertedIndexObject.readFile('book2.json'))).toBe(true);
+  });
+
+  it('should return false for an invalid JSON array', () => {
+    expect(invertedIndexObject.isValidJSON(invertedIndexObject.readFile('invalidJsonBook.json'))).toBe(false);
   });
+});
+
+describe('Validity of file name', () => {
+  it('should return true for a valid file', () => {
+    expect(invertedIndexObject.isValidFileName('book2.json')).toBe(true);
+  });
+
+  it('should return false for an invalid file', () => {
+    expect(invertedIndexObject.isValidFileName('invalidBook.txt')).toBe(false);
+  });
+});
+
+describe('Validity of the index created', () => {
+  it('should return true for a valid index', () => {
+    expect(invertedIndexObject.isIndexValid(invertedIndexObject.createIndex('book1.json', invertedIndexObject.readFile('book1.json')))).toBe(true);
+  });
+
+  it('should return false for an invalid index', () => {
+    expect(invertedIndexObject.isIndexValid(invertedIndexObject.createIndex('invalidJsonBook.json', invertedIndexObject.readFile('invalidJsonBook.json')))).toBe(false);
+  });
+});
+
+describe('create index endpoint', () => {
```

Comment:
We always advice seperating concerns and you have a done a good jod with your test, however, you can better seperate instances by placing API test and Class test in different files